### PR TITLE
MAINT: pysat deprecations

### PR DESCRIPTION
**Pull request by jklenzing**
# Description

Addresses #39

pysat 3.2.0 will drop support for 3 of the test instruments.
- Keeps tests where behaviour is migrated
- Deletes tests where data type (2d pandas) is no longer supported
- Generalize the tests for when dimension lengths in the test instruments change

Also, set the RC install test to "workflow_dispatch', which appears to be an oversight from a previous pull.

## Type of change

- Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

via pytest locally and at GA

**Test Configuration**:
* Operating system: Monterrey
* Version number: Python 3.10.9
* Any details about your local setup that are relevant: pysat branches for main, develop and https://github.com/pysat/pysat/pull/1143

# Checklist:

- [x] Make sure you are merging into the ``develop`` (not ``main``) branch
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] Add a note to ``CHANGELOG.md``, summarizing the changes

 If this is a release PR, replace the first item of the above checklist with the
 release checklist on the pysat wiki:
 https://github.com/pysat/pysat/wiki/Checklist-for-Release

### CODE REVIEW

**Comment by jklenzing** [comment]
Going to use this as the basis for RC testing

**Comment by jklenzing** [comment]
Tests currently failing because pysat 3.1.0 does not have the new kwargs for compatibility. Suggest we leave this until after the 3.2.0 release and add a minimum limit.

Tests with the RC are here: https://github.com/pysat/pysatSeasons/actions/runs/7052829433

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -10,12 +10,12 @@ jobs:
...
       matrix:
-        python-version: ["3.9", "3.10"]
+        python-version: ["3.10", "3.11"]
         os: [ubuntu-latest, macos-latest, windows-latest]
         numpy_ver: [latest]
         include:
-           - python-version: "3.8"
-             numpy_ver: "1.20"
+           - python-version: "3.9"
+             numpy_ver: "1.22"
```
Comment (jklenzing) [inline .github/workflows/main.yml:18]:
```suggestion
             numpy_ver: "1.24"
```

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -10,12 +10,12 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        python-version: ["3.9", "3.10"]
+        python-version: ["3.10", "3.11"]
```
Comment (jklenzing) [inline .github/workflows/main.yml:13]:
```suggestion
        python-version: ["3.11", "3.12"]
```

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -10,12 +10,12 @@ jobs:
...
         os: [ubuntu-latest, macos-latest, windows-latest]
         numpy_ver: [latest]
         include:
-           - python-version: "3.8"
-             numpy_ver: "1.20"
+           - python-version: "3.9"
```
Comment (jklenzing) [inline .github/workflows/main.yml:17]:
```suggestion
           - python-version: "3.10"
```

File: .github/workflows/main.yml [hunk 4]
```diff
@@ -54,9 +52,44 @@ jobs:
...
+    if: always()
+    runs-on: "ubuntu-latest"
+    steps:
+      - name: Coveralls Finish
+        env:
+          COVERALLS_REPO_TOKEN: ${{ secrets.COVERALLS_REPO_TOKEN }}
+          COVERALLS_PARALLEL: true
+        run: |
+          curl -sL https://coveralls.io/coveralls-linux.tar.gz | tar -xz
+          ./coveralls done --build-number ${{ github.run_number }}
```
Comment (aburrell) [inline .github/workflows/main.yml:95]:
```suggestion
          ./coveralls done --build-number ${{ github.run_number }}

```

File: .github/workflows/pip_rc_install.yml [hunk 5]
```diff
@@ -4,22 +4,22 @@
...
+on: [workflow_dispatch]
 
 jobs:
   build:
     strategy:
       fail-fast: false
       matrix:
         os: ["ubuntu-latest", "macos-latest", "windows-latest"]
-        python-version: ["3.10"]  # Keep this version at the highest supported Python version
+        python-version: ["3.11"]  # Keep this version at the highest supported Python version
```
Comment (aburrell) [inline .github/workflows/pip_rc_install.yml:15]:
```suggestion
        python-version: ["3.12"]  # Keep this version at the highest supported Python version
```

**Comment by jklenzing** [comment]
Coveralls fix is downstream of this. Will set up a PR once this is merged.

**Review by aburrell — APPROVED** [review]

**Pull request merged**